node/core/dispute-coordinator/src/real/initialized.rs

```diff
@@ -533,31 +588,45 @@ impl Initialized {
 				}
 			},
 			DisputeCoordinatorMessage::RecentDisputes(tx) => {
-				let recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
-				let _ = tx.send(recent_disputes.keys().cloned().collect());
+				// Return error if session information is missing.
+				if let Some(subsystem_error) = self.error.clone() {
+					return Err(Error::NonFatal(NonFatal::RollingSessionWindow(subsystem_error)))
+				} else {
+					let recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
+					let _ = tx.send(recent_disputes.keys().cloned().collect());
+				}
 			},
 			DisputeCoordinatorMessage::ActiveDisputes(tx) => {
-				let recent_disputes =
-					overlay_db.load_recent_disputes()?.unwrap_or_default().into_iter();
-				let _ =
-					tx.send(get_active_with_status(recent_disputes, now).map(|(k, _)| k).collect());
+				if let Some(subsystem_error) = self.error.clone() {
+					return Err(Error::NonFatal(NonFatal::RollingSessionWindow(subsystem_error)))
+				} else {
+					let recent_disputes =
+						overlay_db.load_recent_disputes()?.unwrap_or_default().into_iter();
+					let _ = tx.send(
+						get_active_with_status(recent_disputes, now).map(|(k, _)| k).collect(),
+					);
+				}
 			},
 			DisputeCoordinatorMessage::QueryCandidateVotes(query, tx) => {
-				let mut query_output = Vec::new();
-				for (session_index, candidate_hash) in query.into_iter() {
-					if let Some(v) =
-						overlay_db.load_candidate_votes(session_index, &candidate_hash)?
-					{
-						query_output.push((session_index, candidate_hash, v.into()));
-					} else {
-						tracing::debug!(
-							target: LOG_TARGET,
-							session_index,
-							"No votes found for candidate",
-						);
+				if let Some(subsystem_error) = self.error.clone() {
+					return Err(Error::NonFatal(NonFatal::RollingSessionWindow(subsystem_error)))
+				} else {
+					let mut query_output = Vec::new();
+					for (session_index, candidate_hash) in query.into_iter() {
```

Comment:
```suggestion
					for (session_index, candidate_hash) in query {
```

`into_iter` is implicitly called in `for` loops

Reply:
Damn, didn't know this fact. Thanks!